tags/for.js

```diff
@@ -38,8 +38,10 @@ module.exports = function (liquid) {
     render: function (scope, hash) {
       var collection = Liquid.evalExp(this.collection, scope)
 
-      if (!Array.isArray(collection) ||
-                (Array.isArray(collection) && collection.length === 0)) {
+      if (isObject(collection)) {
```

Comment:
Can you move these language related functions to [/src/util/underscore.js][underscore-local].

I don't mind renaming `underscore.js` to `lang.js` or something more appropriate if you find this file name confusing, cause it's actually not the [underscore][underscore] libarary.

[underscore-local]: https://github.com/harttle/shopify-liquid/blob/master/src/util/underscore.js
[underscore]: http://underscorejs.org/